server/services/Parser.js

```diff
@@ -1,8 +1,45 @@
-const {MealTimes, Dates} = require('../models/Identifier');
-const StopWord = require('../models/StopWord');
+const { mealTimes, datesLog } = require("../models/Identifier");
+const { stopWordList } = require("../models/StopWord");
 
-class Parser {
-    //TODO Parse through and identify all StopWords & Identifiers
+function parser(string) {
+  const objStringArray = []; // array to store all the matched words objects.
+  const regex = /[`~!@#$%^&*()_|+\-=?;:,'.<>\{\}\[\]\\\/]/gi; //special characters.
+  string = string.replace(regex , "").trim().split(" "); // filter out special characters and spaces.
+
+  for (let i = 0; i < string.length; i++) {
+    let lowerCaseWord = string[i][0].toLowerCase() + string[i].slice(1);
+
+    for (let j = 0; j < mealTimes.length; j++) { // nested loop to check if the word is in mealTimes array or datesLog array.
+      if (mealTimes[j].names.includes(lowerCaseWord)) {
+        objStringArray.push({
+          matchedWord: `${ string[i] }`,
+          type: `meal type`,
+          value: `${ mealTimes[j].value} `,
+        });
+      } else if ( datesLog[j] && datesLog[j].names.includes(lowerCaseWord)) {
```

Comment:
- `mealTimes` and `datesLog` don't share the same length
- If `j = mealTimes.length - 1`, then `datesLog[j]` will be out of bound